### PR TITLE
[ci] Simplify validation test for user model

### PR DESCRIPTION
**Pull request by bgeuken**

### CODE REVIEW

File: src/api/spec/models/user_spec.rb [hunk 1]
```diff
@@ -22,11 +22,7 @@
 
     it { expect(user.state).to eq(User::STATES['unconfirmed']) }
 
-    it 'validates uniqueness of login' do
-      invalid_user = build(:user, login: user.login)
-      invalid_user.save
-      expect(invalid_user.errors.full_messages).to eq(['Login is the name of an already existing user.'])
-    end
+    it { is_expected.to validate_uniqueness_of(:login).with_message('is the name of an already existing user.') }
```
Comment (ChrisBr) [inline src/api/spec/models/user_spec.rb]: Huhh? I also tried this and it didn't worked for me ...

Comment (ChrisBr) [inline src/api/spec/models/user_spec.rb]: Just checked it ... works! Very strange!

Anyway, LGTM :+1: 
Merge as soon as travis is done

Comment (bgeuken) [inline src/api/spec/models/user_spec.rb]: I got this error now: https://travis-ci.org/bgeuken/open-build-service/jobs/111499790

"
Failure/Error:
       project.relationships.create(user: self, role: Role.find_by_title('maintainer'))

     ActiveRecord::RecordNotSaved:
       You cannot call create unless the parent is saved
"

Might be a race condition. Gone try to get around this

**Comment by mdeniz** [comment]
LGTM :+1:

**Pull request merged**